channeldb/invoice_preimages.go

```diff
@@ -9,24 +9,34 @@ import (
   "github.com/lightningnetwork/lnd/extpreimage"
 )
 
+// InvoiceTerm is an interface defining terms that can be
+// used to retrieve hashes and preimages. It is the interface
+// implemented by channeldb.ContractTerm. It is defined as
+// an interface so that it can be stubbed for testing.
+type InvoiceTerm interface {
+  GetPaymentHash() ([32]byte, error)
+  GetPaymentPreimage(uint32, uint32, extpreimage.Client, InvoiceRegistry) (
+    [32]byte, TempPreimageError, PermPreimageError)
+}
+
 // InvoiceRegistry is a registry for storing invoices. It is a
 // simplified interface of htlcswitch.InvoiceDatabase for use
 // in this package.
 type InvoiceRegistry interface {
   AddInvoicePreimage(chainhash.Hash, [32]byte) error
 }
 
-// tempPreimageError is an error encountered while retrieving
+// TempPreimageError is an error encountered while retrieving
 // a preimage which is temporary - we may be able to eventually
 // recover the preimage, but it is in an unknown state.
-type tempPreimageError interface {
+type TempPreimageError interface {
```

Comment:
woops. my bad. should have caught this in the earlier PR. Did linting produce an error for this change?

Reply:
This change is to export the interface so we can use it in testing outside of this package - nothing to catch earlier